Sources/Classes/Models/Configuration.swift

```diff
@@ -18,11 +18,11 @@ public struct Configuration {
     var flushInterval: Int
     var dbCountThreshold: Int
     
-    public init(logLevel: LogLevel, writeKey: String, sdkVersion: String, sdkMetricsUrl: String? = nil, maxErrorsInBatch: Int? = nil, maxMetricsInBatch: Int? = nil, flushInterval: Int? = nil, dbCountThreshold: Int? = nil) {
+    public init(logLevel: LogLevel, writeKey: String, sdkVersion: String, sdkMetricsUrl: String, maxErrorsInBatch: Int? = nil, maxMetricsInBatch: Int? = nil, flushInterval: Int? = nil, dbCountThreshold: Int? = nil) {
         self.logLevel = logLevel
         self.writeKey = writeKey
         self.sdkVersion = sdkVersion
-        self.sdkMetricsUrl = sdkMetricsUrl ?? Constants.Config.SDKMETRICS_URL
+        self.sdkMetricsUrl = sdkMetricsUrl
```

Comment:
The metrics URL is supposed to be automatically deduced based on the data plane URL.
I'm not seeing this logic anywhere.

Reply:
The user already specifies the data plane URL. The metrics URL can be automatically deduced from it, `<dataPlaneUrl>/rsaMetrics`.

@1abhishekpandey, isn't this how we've implemented it in Android SDK as well?

Reply:
We are following this:

1. The user provides the dataPlaneUrl while initialising the native (Android/iOS) SDKs.
2. Our SDK internally take this URL and pass it to the `metrics` module.
3. Metrics module add the path i.e., `rsaMetrics` into that URL.
4. All the new data (error/metrics) are forwarded to that new URL.

Coming to this:
> The user already specifies the data plane URL. The metrics URL can be automatically deduced from it, <dataPlaneUrl>/rsaMetrics.

Yes, the user indeed specifies the dataPlaneURL while SDK init. Next, SDK needs to construct the metrics URL based on the provided dataPlaneUrl. We do that by sending the URL to the metrics module, over there we append the path. That's how metrics URL is constructed (or deduced). This is what we follow in both Android and iOS.

Please LMK if we are missing something?

Reply:
@1abhishekpandey, that's the expectation.

> User has to pass the data plane url as sdkMetricsurl when initializing this SDK.

Then, why is this needed as per @SKannaniOS? 
I'm also asking this because I didn't find the code in this PR where the data plane URL + metrics endpoint are joined together to form the final URL.

The bottom line is that the user should not supply SDK metrics URL input. It should be automatically deduced.

Reply:
Just to clarify a few things:

1. We haven't introduced any new API for the user to use. They only need to pass the dataPlane URL while SDK init.
2. Our native mobile SDK will internally send that same URL to the metrics module for further processing.

Regarding this:
> I didn't find the code in this PR where the data plane URL + metrics endpoint are joined together to form the final URL

- Refer [here](https://github.com/rudderlabs/metrics-reporter-ios/blob/f1bcee2681be2a95ff4cc2df75c373b44343fe44/Sources/Classes/Networking/APIClient/ServiceManager.swift#L127), we are appending the path.
- @SKannaniOS Can provide the exact location, where the above method is being called from the metrics module.
- We will raise another PR in iOS-v1, where we will pass the dataPlaneURL that we receive from the user to this metrics module.

>The bottom line is that the user should not supply SDK metrics URL input. It should be automatically deduced.

The user is not going to provide any configuration, related to the metrics URL. You can review these two Android PRs to get more clarity: [here](https://github.com/rudderlabs/rudder-sdk-android/pull/472/files#diff-35e3689585b9c3d8ec80356aa10e554b0a4fff478e609bdfe3a43f03c5c0e082) and [here](https://github.com/rudderlabs/rudder-sdk-android/pull/471/files).

Please LMK if you need any additional clarification.

Reply:
Sounds good. Thanks.